src/storage.rs

```diff
@@ -1298,15 +1298,15 @@ impl ComponentStorage {
         target: &mut ComponentStorage,
         index: ComponentIndex,
     ) -> Option<Entity> {
-        debug_assert!(index.0 < self.len());
+        debug_assert!(*index < self.len());
         debug_assert!(!target.is_full());
         if !target.is_allocated() {
             target.allocate();
         }
 
-        trace!(index = index.0, source = ?self.id, destination = ?target.id, "Moving entity");
+        trace!(index = *index, source = ?self.id, destination = ?target.id, "Moving entity");
 
-        let entity = unsafe { *self.entities.get_unchecked(index.0) };
+        let entity = unsafe { *self.entities.get_unchecked(*index) };
```

Comment:
This looks like there should be an `ComponentStorage::entity_unchecked(index: ComponentIndex)` since in most cases you don't have to unwrap the index type explicitly to index but instead rely on helper methods.

Reply:
From what I've seen this pattern is only being used twice, so I don't think there is a need for such a function, at least not for now.